src/kafka-net/BrokerRouter.cs

```diff
@@ -155,23 +185,23 @@ private TopicSearchResult SearchCacheForTopics(IEnumerable<string> topics)
             return result;
         }
 
-        private Topic GetCachedTopic(string topic)
+        private Topic GetCachedTopic(string topic, TimeSpan? expiration = null)
         {
-            Topic cachedTopic;
-            return _topicIndex.TryGetValue(topic, out cachedTopic) ? cachedTopic : null;
+            Tuple<Topic, DateTime> cachedTopic;
+            if (_topicIndex.TryGetValue(topic, out cachedTopic))
+            {
+                bool notExistOrExpire = !expiration.HasValue || (DateTime.Now - cachedTopic.Item2).TotalMilliseconds < expiration.Value.TotalMilliseconds;
```

Comment:
Not sure if I understood, if this is true then the data does not exist or expired, so why are you returning the cachedTopic if it's true